enforcer/connection/connection.go

```diff
@@ -142,75 +136,33 @@ func (c *TCPConnection) GetState() TCPFlowState {
 func (c *TCPConnection) SetState(state TCPFlowState) {
 
 	c.state = state
-
-	if TraceLogging == 0 {
-		return
-	}
-
-	c.logs = append(c.logs, fmt.Sprintf("set-state: %s %d", c.String(), state))
 }
 
 // SetReported is used to track if a flow is reported
 func (c *TCPConnection) SetReported(flowState bool) {
 
 	c.flowReported++
 
-	state := ""
-	if c.flowReported > 1 {
-		state = fmt.Sprintf("%t %t", c.flowLastReporting, flowState)
-		zap.L().Debug("Connection reported multiple times",
-			zap.String("state", state))
+	if c.flowReported > 1 && c.flowLastReporting != flowState {
+		zap.L().Info("Connection reported multiple times",
+			zap.Int("report count", c.flowReported),
+			zap.Bool("previous", c.flowLastReporting),
+			zap.Bool("next", flowState),
+		)
 	}
 
 	c.flowLastReporting = flowState
-
-	if TraceLogging == 0 {
-		return
-	}
-
-	// Logging information
-	reported := "flow-reported:"
-	if c.flowReported > 1 {
-		reported = reported + " (ERROR duplicate reporting) " + state
-	}
-
-	if flowState {
-		reported = reported + " dropped: "
-	} else {
-		reported = reported + " accepted: "
-	}
-	reported = reported + c.String()
-
-	c.logs = append(c.logs, reported)
-}
-
-// SetPacketInfo is used to setup the state for the TCP connection
-func (c *TCPConnection) SetPacketInfo(flowHash, tcpFlags string) {
-
-	if TraceLogging == 0 {
-		return
-	}
-
-	pktLog := fmt.Sprintf("pkt-registered: [%s] tcpf:%s %s", flowHash, tcpFlags, c.String())
-	c.logs = append(c.logs, pktLog)
 }
 
 // Cleanup will provide information when a connection is removed by a timer.
 func (c *TCPConnection) Cleanup(expiration bool) {
-
-	logStr := ""
-	for i, v := range c.logs {
-		logStr = logStr + fmt.Sprintf("[%05d]: %s\n", i, v)
-	}
 	// Logging information
-	if c.flowReported == 0 && len(c.logs) > 1 {
+	if c.flowReported == 0 {
 		zap.L().Error("Connection not reported",
-			zap.String("connection", c.String()),
-			zap.String("logs", logStr))
+			zap.String("connection", c.String()))
 	} else {
 		zap.L().Debug("Connection reported",
```

Comment:
do we really need the debug message here?

Reply:
yes. trying to debug duplicate reporting.